internal/controller/storageclaim_controller.go

```diff
@@ -126,7 +127,19 @@ func (r *StorageClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			),
 			builder.OnlyMetadata,
 		).
-		Owns(&csiopv1a1.ClientProfile{}, builder.WithPredicates(generationChangePredicate))
+		Owns(&csiopv1a1.ClientProfile{}, builder.WithPredicates(generationChangePredicate)).
+		Watches(
+			&extv1.CustomResourceDefinition{},
+			&handler.EnqueueRequestForObject{},
+			builder.WithPredicates(
+				utils.NamePredicate(volumeGroupSnapshotClassCrd),
+				utils.CrdCreateAndDeletePredicate(&r.log, volumeGroupSnapshotClassCrd, r.AvailableCrds[volumeGroupSnapshotClassCrd]),
+			),
+			builder.OnlyMetadata,
+		)
```

Comment:
Add this predicate in the previous watch of the CRD
```suggestion
		Watches(
			&extv1.CustomResourceDefinition{},
			&handler.EnqueueRequestForObject{},
			builder.WithPredicates(
				utils.NamePredicate(volumeGroupSnapshotClassCrd),
				utils.CrdCreateAndDeletePredicate(&r.log, volumeGroupSnapshotClassCrd, r.AvailableCrds[volumeGroupSnapshotClassCrd]),
			),
			builder.WithPredicates(
				utils.NamePredicate(drClusterConfigCRDName),
				utils.CrdCreateAndDeletePredicate(&r.log, drClusterConfigCRDName, r.AvailableCrds[drClusterConfigCRDName]),
			),
			builder.OnlyMetadata,
		)
```